node/validator_store.go

```diff
@@ -0,0 +1,540 @@
+package node
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/kvstore"
+	"github.com/Layr-Labs/eigenda/common/kvstore/tablestore"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/litt"
+	"github.com/Layr-Labs/eigenda/litt/littbuilder"
+	"github.com/Layr-Labs/eigenda/litt/util"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	BatchHeaderTableName     = "batch_headers"
+	BlobCertificateTableName = "blob_certificates"
+	BundleTableName          = "bundles"
+	MigrationTableName       = "migration"
+
+	// LevelDBPath is he path where the levelDB database is stored.
+	LevelDBPath = "chunk_v2"
+	// LevelDBDeletionPath is the path where the levelDB database is stored while it is being deleted (for atomicity).
+	LevelDBDeletionPath = "chunk_v2_deleted"
+	// LittDBPath is the path where the littDB database is stored.
+	LittDBPath = "chunk_v2_litt"
+)
+
+// BundleToStore is a struct that holds the bundle key and the bundle bytes.
+type BundleToStore struct {
+	// A bundle key, as encoded by BundleKey()
+	BundleKey []byte
+	// The binary bundle bytes.
+	BundleBytes []byte
+}
+
+// ValidatorStore encapsulates the database for storing batches of chunk data for the V2 validator node.
+type ValidatorStore interface {
+
+	// StoreBatch stores a batch and its raw bundles in the database. Returns the keys of the stored data
+	// and the size of the stored data, in bytes.
+	StoreBatch(batchHeaderHash []byte, batchData []*BundleToStore) ([]kvstore.Key, uint64, error)
+
+	// DeleteKeys deletes the keys from local storage.
+	//
+	// All modifications to the database within this method are performed atomically.
+	DeleteKeys(keys []kvstore.Key) error
+
+	// GetBundleData returns the chunks of a blob with the given bundle key.
+	// The returned chunks are encoded in bundle format.
+	GetBundleData(bundleKey []byte) ([]byte, error)
+
+	// Stop stops the store.
+	Stop() error
+}
+
+type validatorStore struct {
```

Comment:
Would it make more sense to separate the levelDB impl from the littDB impl of `ValidatorStore`?